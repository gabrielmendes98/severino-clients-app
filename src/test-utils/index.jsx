/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react-native';
import MainProvider from 'common/contexts/Main';

const Wrapper = ({ children }) => <MainProvider>{children}</MainProvider>;

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
