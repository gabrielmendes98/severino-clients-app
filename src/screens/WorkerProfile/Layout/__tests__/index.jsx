import React from 'react';
import { View } from 'react-native';
import { rawRender } from 'test-utils';
import WorkerProfileLayout from '../index';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    goBack: jest.fn(),
  }),
}));

it('should match snapshot', () => {
  const { toJSON } = rawRender(
    <WorkerProfileLayout>
      <View></View>
    </WorkerProfileLayout>,
  );

  expect(toJSON()).toMatchSnapshot();
});
