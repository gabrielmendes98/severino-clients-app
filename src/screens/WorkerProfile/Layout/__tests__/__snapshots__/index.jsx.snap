// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7D3CFF",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "padding": 8,
        }
      }
    >
      <View
        color="#FFF"
        name="chevron-left"
        size={24}
      />
    </View>
  </View>
  <RCTScrollView
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "marginBottom": 12,
              "marginTop": 12,
              "paddingHorizontal": 12,
            },
            undefined,
          ]
        }
      >
        <View />
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
